service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -293,35 +288,33 @@ func (r *Resource) newAppCatalogEntries(ctx context.Context, cr v1alpha1.AppCata
 	return entryCRs, nil
 }
 
-func parseLatestVersion(entries []entry) (string, error) {
+func sortEntry(entries []entry) ([]entry, error) {
 	if len(entries) == 0 {
-		return "", nil
+		return nil, nil
 	}
 
-	versions := make([]*semver.Version, len(entries))
-
-	for i, entry := range entries {
-		v, err := semver.NewVersion(entry.Version)
+	sort.Slice(entries, func(i, j int) bool {
+		prevRelease, err := entries[i].Version.SetPrerelease("")
```

Comment:
Remove the prerelease part because it's only SHA and make it impossible to compare between two versions.

Reply:
What about adding a comment for this? So we don't forget why it's needed.